src/algebra/monoid_algebra.lean

```diff
@@ -350,7 +350,7 @@ In particular this provides the instance `algebra k (monoid_algebra k G)`.
 -/
 instance {A : Type*} [comm_semiring k] [semiring A] [algebra k A] [monoid G] :
   algebra k (monoid_algebra A G) :=
-{ smul_def' := λ r a, by { ext, simp [single_one_mul_apply, algebra.smul_def''], },
+{ smul_def' := λ r a, by { ext, simp [single_one_mul_apply, algebra.smul_def'', pi.smul_apply], },
```

Comment:
I didn't know that sometimes it's be necessary to explicitly specify a lemma already tagged with `simp`. Does this increase the priority somehow?

Reply:
Hmm, I've no idea why this is needed. I guess I should double check that it really is.

Reply:
I checked before I posted my comment; when I removed `pi.smul_apply` there the goal ends up being:
```
⊢ ⇑(algebra_map k (has_coe_to_fun.F (r • a))) r a_1 * ⇑a a_1 = ⇑(algebra_map k A) r * ⇑a a_1
```

Reply:
This means that the simp set isn't confluent right? Can we identify some missing simp lemma in that goal?